sentry_sdk/integrations/strawberry.py

```diff
@@ -0,0 +1,401 @@
+import hashlib
+from functools import cached_property
+from inspect import isawaitable
+from sentry_sdk import configure_scope, start_span
+from sentry_sdk.consts import OP
+from sentry_sdk.integrations import Integration, DidNotEnable
+from sentry_sdk.integrations.logging import ignore_logger
+from sentry_sdk.integrations.modules import _get_installed_modules
+from sentry_sdk.hub import Hub, _should_send_default_pii
+from sentry_sdk.utils import (
+    capture_internal_exceptions,
+    event_from_exception,
+    logger,
+    parse_version,
+)
+from sentry_sdk._types import TYPE_CHECKING
+
+try:
+    import strawberry.schema.schema as strawberry_schema  # type: ignore
+    from strawberry import Schema
+    from strawberry.extensions import SchemaExtension  # type: ignore
+    from strawberry.extensions.tracing.utils import should_skip_tracing  # type: ignore
+    from strawberry.extensions.tracing import (  # type: ignore
+        SentryTracingExtension as StrawberrySentryAsyncExtension,
+        SentryTracingExtensionSync as StrawberrySentrySyncExtension,
+    )
+    from strawberry.http import async_base_view, sync_base_view  # type: ignore
+except ImportError:
+    raise DidNotEnable("strawberry-graphql is not installed")
+
+if TYPE_CHECKING:
+    from typing import Any, Callable, Dict, Generator, List, Optional
+    from graphql import GraphQLError, GraphQLResolveInfo  # type: ignore
+    from strawberry.http import GraphQLHTTPResponse
+    from strawberry.types import ExecutionContext, ExecutionResult  # type: ignore
+    from sentry_sdk._types import EventProcessor
+
+
+ignore_logger("strawberry.execution")
+
+
+class StrawberryIntegration(Integration):
+    identifier = "strawberry"
+
+    def __init__(self, async_execution=None):
+        # type: (Optional[bool]) -> None
+        if async_execution not in (None, False, True):
+            raise ValueError(
+                'Invalid value for async_execution: "{}" (must be bool)'.format(
+                    async_execution
+                )
+            )
+        self.async_execution = async_execution
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        installed_packages = _get_installed_modules()
+        version = parse_version(installed_packages["strawberry-graphql"])
```

Comment:
unfortunately we don't have it :(

Reply:
I'd like to move to using `_get_installed_modules` to get the version in all our integrations eventually since `importlib.metadata` (where it takes the data from) is generic, works for all pypi packages, and we're not dependent on a usually undocumented constant that, if moved, would break the whole integration.

Reply:
Okay, fair enough @sentrivana. However, in that case, I think maybe it would be worthwhile to extract this logic to a separate method that would retrieve and parse the version of a given package in one call. Basically, my idea would be to replace the above code with something like the following:
```python
version = _get_version_of_module("strawbery-graphql")
```

Reply:
That's a good idea. Since we need this in multiple places this is a bit out of scope for this PR, so I've opened a separate PR: https://github.com/getsentry/sentry-python/pull/2429 (I will add strawberry there as soon as this is merged).